src/js/tech/ChromecastTech.js

```diff
@@ -0,0 +1,364 @@
+'use strict';
+
+var ChromecastSessionManager = require('../chromecast/ChromecastSessionManager'),
+    ChromecastTechUI = require('./ChromecastTechUI'),
+    _ = require('underscore'),
+    SESSION_TIMEOUT = 10 * 1000, // milliseconds
+    ChromecastTech;
+
+
+ChromecastTech = {
+   constructor: function(options) {
+      var subclass;
+
+      this._eventListeners = [];
+
+      // We have to initialize the UI here, before calling super.constructor
+      // because the constructor calls `createEl`, which references `this._ui`.
+      this._ui = new ChromecastTechUI();
+      this._ui.updatePoster(options.source.poster);
+
+      // Call the super class' constructor function
+      subclass = this.constructor.super_.apply(this, arguments);
+
+      this._chromecastSessionManager = options.source.chromecastSessionManager;
+      this._remotePlayer = this._chromecastSessionManager.getRemotePlayer();
+      this._remotePlayerController = this._chromecastSessionManager.getRemotePlayerController();
+      this._listenToPlayerControllerEvents();
+      this.on('dispose', this._removeAllEventListeners.bind(this));
+
+      this._hasPlayedAnyItem = false;
+      this._currentSource = options.source;
+      this._requestTitle = options.requestTitleFn || _.noop;
+      this._requestSubtitle = options.requestSubtitleFn || _.noop;
+      // See `currentTime` function
+      this._firstItemStartTime = options.startTime || 0;
+      // See `duration` function
+      this._firstItemDuration = options.source.duration || 0;
```

Comment:
However, does it matter that other sources likely won't have the duration? Once `_hasPlayedAnyItem` is true, we no longer need to look at the source for the duration. If this plugin is always setting the initial source, there shouldn't be an issue.

As it currently stands, if something else changes the source before an item was played, the code will report the wrong duration. By using `duration` from the source in the `duration` function, this would at least provide a user-facing work-around to this.